### DISCUSSION TITLE
Can attackers bypass restrict mode 4 if they obfuscate their scripts?

### DISCUSSION THREAD

**Discussion by Kexiii** [dashingsoft/pyarmor#1222] [Q&A] [answered]
If script is obfuscated by restrict mode 4, it will raise protection exception if the caller is not obfuscated.
My question is, will it also be able to protect the script if some attacker obfuscates their malicious script?
What's the mechanism behind restrict mode 4? Will it check some signature to make sure the caller is obfuscated by the same owner? 
Thanks.

**Comment by jondy** [comment, marked as answer]
Yes. Only your own obfuscated script is valid, other's obfuscated scripts are invalid.

**Reply by Kexiii** [reply]
How is it implemented? If I and my coworker share the same development linux host(on which I obfuscated my scripts), will he be able to access my scripts?

**Reply by jondy** [reply]
> How is it implemented? 

Sorry, I can't tell you, :)

You can verify it by yourself

**Reply by Kexiii** [reply]
ok, thanks again for quick reply
